Get topics #2

ojcorrigan opened the pull request:
Rather large pull request:
set up app.js, 
created test file, 
created controller and model files, 
used TDD to create get on endpoint /api/topics which works
added generic error handling for 500 server error, 
404 seems to be handled automatically I think but may be missing something.

owen-corrigan committed:
created app.js to start my server

owen-corrigan committed:
created controllers and models with a view to creating my get endpoint

owen-corrigan committed:
set up first test for app.js

owen-corrigan committed:
rather large commit but get for endpoint solved task #2

giteron reviewed (changes requested):
Good work so far! A couple of amends I would suggest:
- GET /api/topics should return topics in an object, where the key is ‘topics’ and the value is your array of topics
- In the test, if the array was much larger, we would want to be more dynamic with testing each 'topic' object in it. How could we do that? I would suggest implementing this here.

Other considerations:
- Consider why 404s are currently working automatically, and whether you are happy with this! (I would advise explicitly handling it)
- In your test description, good to mention what the endpoint responds with an array of
- Remember to clean up console.logs as you go along :)

owen-corrigan committed:
created specific controller to handle invalid path

ojcorrigan commented:
Have updated:
 cleaned up console.logs, 
updated tests to specify the end point being tested 
created a specific function in a new controllers file to handle invalid file paths that has a custom message
